.github/workflows/publish.yml

```diff
@@ -37,6 +37,9 @@ jobs:
 
       - name: Publish docker image
         run: ./gradlew publish dockerPushImages
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
```

Comment:
write credentials should be kept distinct from read credentials (this will be a read-only token once dockerhub allows assigning permission to tokens)